### PR TITLE
pkg/asset/installconfig/openstack: Cleaner and Tested Validation Code

### PR DESCRIPTION
**Pull request by iamemilio**
Add unit tests and defaults to OpenStack Validations, and refactor for easier testing and reading

/cc @Fedosin @abhinavdahiya

### CODE REVIEW

**Comment by Fedosin** [comment]
@abhinavdahiya agree! I'm working on the patch that fixes it.

**Comment by iamemilio** [comment]
ACK removed

File: pkg/asset/installconfig/openstack/validate.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import (
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/openshift/installer/pkg/asset/installconfig/openstack/validation"
+	"github.com/openshift/installer/pkg/types"
+)
+
+var (
+	validFlavor = "valid-flavor"
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/openstack/validate.go:11]: These look like const for testing.. keep them in _test.go

Comment (iamemilio) [reply pkg/asset/installconfig/openstack/validate.go:11]: oh oops. good catch

**Comment by iamemilio** [comment]
seems like the openstack test infra is having trouble. Machines are failing to come up and installs are timing out.

**Comment by iamemilio** [comment]
@sdodson tthis pr is blocked by aws, metal and crc tests but only alters openstack code. Can we override? This blocks a lot of openstack work

**Comment by sdodson** [comment]
/override ci/prow/e2e-aws
AWS job is busted right now and this only affects openstack code

**Pull request merged**